Bottom: 70fcd43d7ae1ef5f8cbc6819753c088454135b33
Top:    babb1a79a47215da54457545ecc8e1353c64d1ab
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-02-15 19:47:54 +0000

Tmp changes


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6abd4ff..e2a7255 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1813,16 +1813,19 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
   VNODE *cur;
   DIRECTION dir = UNINITIALISED;
 
-  cur = (&a->head); //    ->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
+//  cur = (&a->head)->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
+//  cur = (&a->head); //    ->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
   cur = (&a->head)        ;
   do
     {
+#if 0
       // The following may be a nice speedup, but not sure if it is correct.
       // In particular, consider the case when we collect a 'B' polygon contour.
       // Could some of that countour may already have been collected, and there
       // still be a piece we are interested in after? (Can we reach it though??)
       if (cur->Flags.mark != 0)
         break;
+#endif
 
       if (j_rule (poly, cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
